src/DataValidator.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace League\ISO3166;
+
+class DataValidator
+{
+    use KeyValidators;
+
+    /**
+     * @param array $data
+     *
+     * @return array
+     */
+    public function validate(array $data)
+    {
+        foreach ($data as $entry) {
+            $this->assertEntryHasRequiredKeys($entry);
+        }
+
+        return $data;
+    }
+
+    /**
+     * @param array $entry
+     *
+     * @throws \DomainException if given data entry does not have all the required keys.
+     */
+    private function assertEntryHasRequiredKeys(array $entry)
```

Comment:
Yeah I'm with @frankdejonge on this; the less we make public, the less we constrain ourselves in regards to future changes.